aten/src/ATen/native/PixelShuffle.cpp

```diff
@@ -4,31 +4,45 @@
 #include <c10/util/Exception.h>
 
 #include <algorithm>
+#include <numeric>
 #include <vector>
 
 namespace at {
 namespace native {
 
 Tensor pixel_shuffle(const Tensor& self, int64_t upscale_factor) {
-  AT_ASSERTM(self.dim() == 4,
-             "pixel_shuffle expects 4D input, but got input with sizes ",self.sizes());
-  int64_t b = self.size(0);
-  int64_t c = self.size(1);
-  int64_t h = self.size(2);
-  int64_t w = self.size(3);
+  TORCH_CHECK(self.dim() >= 3,
+             "pixel_shuffle expects at least 3D input, but got input with sizes ", self.sizes());
```

Comment:
This error message is not a regression, but I think it could be slightly improved.

First, the check is for dimensionality, but when this check fails the sizes are shown to the user. It seems like it'd be more consistent to say the input had X dims to be consistent with the check.

Second, the use of "3D" here may be misleading, as pixel shuffle shuffles batches of (conceptually) 3D tensors. I think this can also be addressed by consistently referencing dims, so the error message would be something like:

```
"... expects input to have at least 3 dimensions, but got input with X dimensions"
```